File: java/src/br/unifor/facilitador/dominio/Documento.java
package br.unifor.facilitador.dominio;

import java.time.LocalDateTime;
import java.util.UUID;

/** Exame previo, laudo, receita ou atestado trafegado entre paciente e medico. */
public class Documento {

    private final UUID id = UUID.randomUUID();
    private final TipoDocumento tipo;
    private final String uri;
    private final LocalDateTime enviadoEm = LocalDateTime.now();
    private final UUID autor;

    public Documento(TipoDocumento tipo, String uri, UUID autor) {
        this.tipo = tipo;
        this.uri = uri;
        this.autor = autor;
    }

    public UUID getId() { return id; }
    public TipoDocumento getTipo() { return tipo; }
    public String getUri() { return uri; }
    public LocalDateTime getEnviadoEm() { return enviadoEm; }
    public UUID getAutor() { return autor; }

    @Override
    public String toString() {
        return tipo + " (" + uri + ")";
    }
}

File: javascript/src/dominio/Documento.js
import { randomUUID } from 'node:crypto';

// Exame previo, laudo, receita ou atestado trafegado entre paciente e medico.
export class Documento {
  #id = randomUUID();
  #tipo;
  #uri;
  #enviadoEm = new Date();
  #autor;

  constructor(tipo, uri, autor) {
    this.#tipo = tipo;
    this.#uri = uri;
    this.#autor = autor;
  }

  get id() { return this.#id; }
  get tipo() { return this.#tipo; }
  get uri() { return this.#uri; }
  get enviadoEm() { return this.#enviadoEm; }
  get autor() { return this.#autor; }

  toString() {
    return `${this.#tipo} (${this.#uri})`;
  }
}
